library/objective-c/EnvoyBridgeUtility.h

```diff
@@ -100,6 +100,23 @@ static inline NSString *to_ios_string(envoy_data data) {
   return platformString;
 }
 
+static inline EnvoyEvent *to_ios_map(envoy_map map) {
+  NSMutableDictionary *newMap = [NSMutableDictionary new];
+  for (envoy_map_size_t i = 0; i < map.length; i++) {
+    envoy_map_entry header = map.entries[i];
+    NSString *headerKey = [[NSString alloc] initWithBytes:header.key.bytes
+                                                   length:header.key.length
+                                                 encoding:NSUTF8StringEncoding];
+    NSString *headerValue = [[NSString alloc] initWithBytes:header.value.bytes
+                                                     length:header.value.length
+                                                   encoding:NSUTF8StringEncoding];
+    newMap[headerKey] = headerValue;
+  }
+
+  release_envoy_map(map);
+  return [newMap copy];
```

Comment:
Do we actually need a copy here? Looks like `to_ios_headers` does not copy // @goaway

Reply:
we do not need it per se - I just decided to make a copy so that we return an instance of `NSDictionary` as opposed to `NSMutableDictionary`. Without copying here it's possible for a caller of this method to take the returned value, cast it to `NSMutableDictionary` and modify it... that being said, it's highly unlike and copying costs resources so I will just get rid of this copying.

Reply:
I'd definitely bias towards reducing the copy here since external consumers would be misusing the API by trying to mutate it